flink-container/src/test/java/org/apache/flink/container/entrypoint/StandaloneJobClusterEntryPointTest.java

```diff
@@ -67,54 +65,109 @@ public void testDontOverwriteExecutionMode() {
+	public void configuredJobIdTakesPrecedenceWithHA() {
+		JobID jobId = JobID.generate();
+		String jobIdSeed = null;
+
+		Configuration globalConfiguration = new Configuration();
+		enableHighAvailability(globalConfiguration);
+
+		JobID jobIdForCluster = StandaloneJobClusterEntryPoint.resolveJobIdForCluster(
+			jobId,
+			jobIdSeed,
+			globalConfiguration);
+
+		assertThat(jobIdForCluster, is(jobId));
+	}
+
+	@Test
+	public void configuredJobIdTakesPrecedenceWithoutHA() {
+		JobID jobId = JobID.generate();
+		String jobIdSeed = null;
+
+		Configuration globalConfiguration = new Configuration();
+
+		JobID jobIdForCluster = StandaloneJobClusterEntryPoint.resolveJobIdForCluster(
+			jobId,
+			jobIdSeed,
+			globalConfiguration);
+
+		assertThat(jobIdForCluster, is(jobId));
+	}
+
+	@Test
+	public void configuredJobIdSeedTakesPrecedenceWithoutHA() {
+		JobID jobId = null;
+		String jobIdSeed = "some-seed";
+
+		Configuration globalConfiguration = new Configuration();
+
+		JobID jobIdForCluster = StandaloneJobClusterEntryPoint.resolveJobIdForCluster(
+			jobId,
+			jobIdSeed,
+			globalConfiguration);
+
+		assertThat(jobIdForCluster, is(JobID.fromSeed(jobIdSeed)));
+	}
+
+	@Test
+	public void configuredJobIdSeedTakesPrecedenceWithHA() {
+		JobID jobId = null;
+		String jobIdSeed = "some-seed";
 
 		Configuration globalConfiguration = new Configuration();
 		enableHighAvailability(globalConfiguration);
 
 		JobID jobIdForCluster = StandaloneJobClusterEntryPoint.resolveJobIdForCluster(
-			jobID,
+			jobId,
+			jobIdSeed,
 			globalConfiguration);
 
-		assertThat(jobIdForCluster, is(jobID.get()));
+		assertThat(jobIdForCluster, is(JobID.fromSeed(jobIdSeed)));
 	}
 
 	@Test
-	public void configuredJobIDTakesPrecedenceWithoutHA() {
-		Optional<JobID> jobID = Optional.of(JobID.generate());
+	public void configuredJobIdTakesPrecedenceOverJobIdSeed() {
+		JobID jobId = JobID.generate();
+		String jobIdSeed = "some-seed";
 
 		Configuration globalConfiguration = new Configuration();
+		enableHighAvailability(globalConfiguration);
 
 		JobID jobIdForCluster = StandaloneJobClusterEntryPoint.resolveJobIdForCluster(
-			jobID,
+			jobId,
+			jobIdSeed,
 			globalConfiguration);
 
-		assertThat(jobIdForCluster, is(jobID.get()));
+		assertThat(jobIdForCluster, is(jobId));
 	}
 
 	@Test
-	public void jobIDdefaultsToZeroWithHA() {
-		Optional<JobID> jobID = Optional.empty();
+	public void jobIdDefaultsToZeroWithHA() {
+		JobID jobId = null;
+		String jobIdSeed = null;
 
 		Configuration globalConfiguration = new Configuration();
 		enableHighAvailability(globalConfiguration);
 
 		JobID jobIdForCluster = StandaloneJobClusterEntryPoint.resolveJobIdForCluster(
-			jobID,
+			jobId,
+			jobIdSeed,
 			globalConfiguration);
 
 		assertThat(jobIdForCluster, is(ZERO_JOB_ID));
 	}
 
 	@Test
-	public void jobIDdefaultsToRandomJobIDWithoutHA() {
-		Optional<JobID> jobID = Optional.empty();
+	public void jobIdDefaultsToRandomJobIDWithoutHA() {
```

Comment:
Rename `jobIdDefaultsToRandomJobIDWithoutHA` to `jobIdDefaultsToRandomJobIdWithoutHA`, to finish getting rid of `ID` in the method/variables names in this file?

Reply:
Done the renaming. I agree about the side note, but will not do this as part of this PR if that's ok.